### PR TITLE
Nonpr 2162 templates

### PR DESCRIPTION
**Pull request by PaulHodgson**
Use up to date `HMRC` and `gov-uk` components to simplify the templates. See https://github.com/hmrc/play-frontend-hmrc

Tidy the tests too.

This will need to be merged along with the corresponding `nrs_retrieval-acceptance-tests` pr as there are inter-dependencies. See https://github.com/hmrc/nrs-retrieval-acceptance-tests/pull/52

```
sbt test 
...
[info] ScalaTest
[info] Run completed in 11 seconds, 941 milliseconds.
[info] Total number of tests run: 60
[info] Suites: completed 17, aborted 0
[info] Tests: succeeded 60, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 60, Failed 0, Errors 0, Passed 60
[success] Total time: 18 s, completed 29-Jul-2021 15:15:57
```

```
sbt it:test
...
[info] ScalaTest
[info] Run completed in 28 seconds, 278 milliseconds.
[info] Total number of tests run: 25
[info] Suites: completed 4, aborted 0
[info] Tests: succeeded 25, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 25, Failed 0, Errors 0, Passed 25
```

### CODE REVIEW

File: app/views/components/Button.scala.html [hunk 1]
```diff
@@ -14,13 +14,9 @@
  * limitations under the License.
  *@
```
Comment (PaulHodgson) [inline app/views/components/Button.scala.html:16]: not sure why this shows up as a rename. Added a button component.

File: app/views/components/Heading1.scala.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*
```
Comment (PaulHodgson) [inline app/views/components/Heading1.scala.html:1]: again not sure why this is seen as a rename. Added a component for H1

File: app/views/components/Layout.scala.html [hunk 3]
```diff
@@ -0,0 +1,80 @@
+@import views.html.components.Heading1
```
Comment (PaulHodgson) [inline app/views/components/Layout.scala.html:1]: New layout component as a basis for all templates. This works with the latest HMRC and gov_uk components and helpers.

File: app/views/components/Layout.scala.html [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        hmrcInternalHeader: HmrcInternalHeader,
+        hmrcStandardFooter: HmrcStandardFooter,
+        hmrcHead: HmrcHead,
+        hmrcLanguageSelectHelper: HmrcLanguageSelectHelper,
+        hmrcScripts: HmrcScripts,
+        defaultMainContent: TwoThirdsMainContent,
+        h1: Heading1,
+        govukBackLink: GovukBackLink)
+
+@(  pageTitle: String,
```
Comment (PaulHodgson) [inline app/views/components/Layout.scala.html:21]: usually we don't specify most of this stuff. The search page is different.

File: app/views/components/TextInput.scala.html [hunk 5]
```diff
@@ -14,23 +14,15 @@
  * limitations under the License.
```
Comment (PaulHodgson) [inline app/views/components/TextInput.scala.html:14]: Again, this is an add not a rename

File: app/views/error_template.scala.html [hunk 6]
```diff
@@ -15,14 +15,12 @@
...
 
-@this(mainTemplate: views.html.main_template)
+@this(layout:Layout, p: Paragraph)
 
 @(title: String, heading: String, message: String)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
 
-@mainContent = {
-  <h1>@heading</h1>
-  <p>@message</p>
+@layout(pageTitle = title, maybePageHeader = Some(heading)){
```
Comment (PaulHodgson) [inline app/views/error_template.scala.html:24]: templates now all use the new layout component

File: app/views/selector_page.scala.html [hunk 7]
```diff
@@ -18,57 +18,34 @@
...
-  Messages("selector.page.header.lbl")
+    val title = Messages("selector.page.header.lbl")
+    if (selectorForm.hasErrors) s"${Messages("generic.errorPrefix")} $title" else title
 }
 
 @radioOptions = @{
   appConfig.notableEvents.keys.map(key => RadioParams(key, appConfig.notableEvents(key).displayName)).toList.sortBy(_.label)
 }
 
-@mainTemplate(title = titlePrefix + title, userName = userName) {
```
Comment (PaulHodgson) [inline app/views/selector_page.scala.html:44]: we use the standard error handling components here now

File: build.sbt [hunk 8]
```diff
@@ -59,6 +58,14 @@ lazy val root = (project in file("."))
     scalacOptions += "-P:silencer:pathFilters=target/.*",
     publishingSettings,
     scoverageSettings)
+  .settings(
+    TwirlKeys.templateImports ++= Seq(
```
Comment (PaulHodgson) [inline build.sbt:62]: this auto-imports hmrc and govuk components into our templates to save boiler plate

File: conf/app.routes [hunk 9]
```diff
@@ -1,6 +1,8 @@
-# route to serve up local assets
-GET        /assets/*file                            controllers.Assets.versioned(path="/public", file: Asset)
+# assets routes
```
Comment (PaulHodgson) [inline conf/app.routes:1]: we now have assets from 3 places...

File: conf/messages [hunk 10]
```diff
@@ -1,80 +1,43 @@
-service.name.lbl=Search the Non-Repudiation Store
+service.name=Search the Non-Repudiation Store
```
Comment (PaulHodgson) [inline conf/messages:1]: we needed a `service-name` for the component upgrade so got rid of the similar message key

File: conf/messages [hunk 11]
```diff
@@ -1,80 +1,43 @@
...
 
 generic.error=There is a problem
 generic.errorPrefix=Error:
 
 ################################## General #######################################################
```
Comment (PaulHodgson) [inline conf/messages:6]: there were a lot of unused message keys

File: conf/messages [hunk 12]
```diff
@@ -1,80 +1,43 @@
...
-search.results.submitted.lbl=Received at
 search.results.filename.lbl=Filename
-search.results.results.lbl=results
 search.results.retrieve.lbl=Retrieve
 search.results.download.lbl=Download
 search.results.cannot.retrieve.lbl=We can not retrieve
 search.results.retrieval.message.lbl=Retrieving {0}
 search.results.retrieval.info=(this may take up to 5 minutes)
-search.results.error.summary.lbl=There's a problem
+search.results.error.summary.lbl=There''s a problem
```
Comment (PaulHodgson) [inline conf/messages:38]: you need to escape quotes in message files

File: test/actors/ActorSpec.scala [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+import config.AppConfig
+import connectors.NrsRetrievalConnector
+import org.mockito.Mockito.when
+import org.scalatest.BeforeAndAfterAll
+import support.UnitSpec
+
+import java.util.concurrent.TimeUnit.SECONDS
+import scala.concurrent.duration.{FiniteDuration, _}
+
+class ActorSpec extends TestKit(ActorSystem("MySpec")) with ImplicitSender with UnitSpec with BeforeAndAfterAll {
```
Comment (PaulHodgson) [inline test/actors/ActorSpec.scala:32]: abstracted common `akka Actor` requirements here

File: test/controllers/ControllerSpec.scala [hunk 14]
```diff
@@ -16,31 +16,16 @@
...
 import uk.gov.hmrc.auth.core.AuthConnector
-import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 import views.html.error_template
 
-trait ControllerSpec extends GuiceAppSpec with MockitoSugar with StubControllerComponentsFactory with Status {
-
+trait ControllerSpec extends GuiceAppSpec with StubControllerComponentsFactory {
   val getRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest("GET", "/")
   val postRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest("POST", "/")
+  val mockAuthConnector: AuthConnector = mock[AuthConnector]
```
Comment (PaulHodgson) [inline test/controllers/ControllerSpec.scala:29]: most of this stuff was moved upwards.

File: test/support/GuiceAppSpec.scala [hunk 15]
```diff
@@ -17,37 +17,37 @@
...
 
-class GuiceAppSpec extends BaseSpec {
-
-  protected val bindModules: Seq[GuiceableModule] = Seq()
-  private val additionalConfig = Map(
-    "google-analytics.host" -> "host",
-    "google-analytics.token" -> "aToken")
-
-  val mockNrsRetrievalConnector: NrsRetrievalConnector = mock[NrsRetrievalConnector]
+class GuiceAppSpec extends UnitSpec with GuiceOneAppPerSuite with PatienceConfiguration { this: Suite =>
```
Comment (PaulHodgson) [inline test/support/GuiceAppSpec.scala:31]: I merged `BaseSpec` into this as  `BaseSpec` was only extended here.

File: test/support/UnitSpec.scala [hunk 16]
```diff
@@ -16,14 +16,27 @@
...
+import play.api.{Configuration, Environment}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 
-import scala.concurrent.{Await, Future}
 import scala.concurrent.duration.{FiniteDuration, _}
+import scala.concurrent.{Await, Future}
 
-class UnitSpec extends AnyWordSpec with Matchers {
+trait UnitSpec extends AnyWordSpecLike with Matchers with MockitoSugar with Status {
```
Comment (PaulHodgson) [inline test/support/UnitSpec.scala:31]: made this a trait so that it could be used by the Actor Specs. Added useful commons,

File: test/support/fixtures/SearchFixture.scala [hunk 17]
```diff
@@ -16,27 +16,10 @@
...
 
-import config.AppConfig
-import models.{SearchResult, SearchResultUtils}
-import play.api.libs.json.{JsValue, Json}
-import play.api.{Configuration, Environment}
-import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import models.SearchResult
 
 trait SearchFixture extends NrSubmissionId {
-
```
Comment (PaulHodgson) [inline test/support/fixtures/SearchFixture.scala:26]: mostly unused

File: app/views/components/Layout.scala.html [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+@contentBlock() = {
+  @banner
+  @maybeBackLink
+  @h1(id="pageHeader", content = Text(maybePageHeader.getOrElse(pageTitle)))
+  @content
+}
+
+@govukLayout(
```
Comment (PaulHodgson) [inline app/views/components/Layout.scala.html:68]: we have to step down a level from the top level abstractions as we're an internal service so don't want the standard HMRC header.

File: conf/messages (hunk 11 above)
Comment (Leeroyc720) [reply conf/messages:6]: good spotting, I think vat reg was one of the first services to be added before going dynamic

File: test/controllers/RoutesSpec.scala [hunk 19]
```diff
@@ -52,50 +50,53 @@ class RoutesSpec extends GuiceAppSpec with BaseSpec with NrsSearchFixture {
...
           ("notableEventType", notableEventType)
         )))
 
         result.map(status(_)) shouldBe Some(OK)
 
         val text: String = result.map(contentAsString(_)).get
-        text should include(Messages(s"search.page.$notableEventType.header.lbl"))
-        text should not include Messages("search.results.notfound.lbl")
-        text should include(Messages("search.results.results.lbl"))
+        text should include("We found 1 results for VAT Registration Number (VRN): &quot;" + foundVrn + "&quot;")
```
Comment (Leeroyc720) [inline test/controllers/RoutesSpec.scala:93]: Don't we need the not-found test result in this?

Comment (PaulHodgson) [reply test/controllers/RoutesSpec.scala:93]: Seems odd to test for the success message (`we found results`) but then also test that the zero results message is not displayed, so I removed it.

Also this is a RoutesSpec - I don't think it should be making detailed assertions on page content.

Comment (PaulHodgson) [reply test/controllers/RoutesSpec.scala:93]: I might build out `search_pageSpec` to capture this kind of detail instead?

Comment (Leeroyc720) [reply test/controllers/RoutesSpec.scala:93]: I agree, it may not be necessary, if we are covering this somewhere else in the specs

**Review by Leeroyc720 — APPROVED** [review]
Looks good Paul, apart from the design issues we noticed during the demo, regarding the links I think they are good the way you have done them, noticed some are in `messages`

File: test/support/GuiceAppSpec.scala (hunk 15 above)
Comment (Leeroyc720) [reply test/support/GuiceAppSpec.scala:31]: nice

**Review by craigtd — APPROVED** [review]
looks good and all tests passing
